Framework/API/src/InfoComponentVisitor.cpp

```diff
@@ -137,5 +144,9 @@ size_t InfoComponentVisitor::size() const {
   return m_componentIds.size() - m_droppedDetectors;
 }
 
+const std::unordered_map<Mantid::Geometry::IComponent *, size_t> &
+InfoComponentVisitor::componentIdToIndexMap() const {
+  return m_componentIdToIndexMap;
```

Comment:
I just thought that it is an optimization you get for free without even adding an extra line of code. Probably not super relevant though.